Source/santad/ProcessTree/EndpointSecurityAdapter.mm

```diff
@@ -0,0 +1,64 @@
+/// Copyright 2023 Google LLC
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     https://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+#include <EndpointSecurity/EndpointSecurity.h>
+#include <Foundation/Foundation.h>
+#include <bsm/libbsm.h>
+
+#include "Source/santad/ProcessTree/process_tree.h"
+#include "Source/santad/ProcessTree/process_tree_macos.h"
+#include "absl/status/statusor.h"
+
+namespace santa::santad::process_tree {
+
+void InformFromESEvent(ProcessTree &tree, const es_message_t *msg) {
+  struct Pid event_pid = PidFromAuditToken(msg->process->audit_token);
+  auto proc = tree.Get(event_pid);
+
+  if (!proc) {
+    return;
+  }
+
+  switch (msg->event_type) {
+    case ES_EVENT_TYPE_AUTH_EXEC:
+    case ES_EVENT_TYPE_NOTIFY_EXEC: {
+      std::vector<std::string> args;
+      args.reserve(es_exec_arg_count(&msg->event.exec));
```

Comment:
We probably need (or at least need to consider...) for these ES API calls to go through an instance of `EndpointSecurityAPI.h` so that they can be mocked out. There aren't tests here, but once this is hooked into Santa proper and we have tests for the new client (e.g. SNTEndpointSecurityTreeAwareClient.mm), we'll need to make sure we can properly control for these calls.